version: '3'
services:
  redis:
      image: redis
      ports:
        - '6379:6379'

  db:
    image: postgres:14
    environment:
      POSTGRES_DB: ${DATABASES_DEFAULT_NAME}
      POSTGRES_USER: ${DATABASES_DEFAULT_USER}
      POSTGRES_PASSWORD: ${DATABASES_DEFAULT_PASSWORD}

    container_name: db
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports:
      - '5432:5432'
    healthcheck:
      test: [ 'CMD-SHELL', 'pg_isready -U postgres' ]
      interval: 10s
      timeout: 5s
      retries: 5

  app:
    build: .
    tty: true
    command: python manage.py migrate && python manage.py runserver 0.0.0.0:8080
    ports:
      - "8080:8000"
    volumes:
      - .:/habit
      - ./static:/app/static
    depends_on:
      db:
        condition: service_healthy
    env_file:
      - .env

  celery:
    restart: always
    build: .
    command: celery -A config worker —loglevel=info && celery -A config beat —loglevel=info
    volumes:
      - .:/app
    depends_on:
      - app
      - redis

  celery-beat:
    build: .
    restart: always
    tty: true
    command: celery -A config beat -l info -S django
    depends_on:
      - redis
      - app
volumes:
  pg_data:
  static:
